airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/destination/async/AsyncStreamConsumer.kt

```diff
@@ -104,12 +107,43 @@ constructor(
             airbyteMessageDeserializer.deserializeAirbyteMessage(
                 message,
             )
-        if (AirbyteMessage.Type.RECORD == partialAirbyteMessage.type) {
-            validateRecord(partialAirbyteMessage)
-
-            partialAirbyteMessage.record?.streamDescriptor?.let {
-                getRecordCounter(it).incrementAndGet()
+        when (partialAirbyteMessage.type) {
+            AirbyteMessage.Type.RECORD -> {
+                validateRecord(partialAirbyteMessage)
+
+                partialAirbyteMessage.record?.streamDescriptor?.let {
+                    getRecordCounter(it).incrementAndGet()
+
+                    if (streamStatusesFromSource.containsKey(it)) {
```

Comment:
NIT: is the variable poorly named and should be "terminalStreamStatusesFromSource" ?

Reply:
renamed to terminalStatusesFromSource